ReminderApp-OBJC/ReminderApp-OBJC/Classes/Controller/CDReminderTableViewController.m

```diff
@@ -129,62 +101,231 @@ - (void)controller:(NSFetchedResultsController *)controller didChangeObject:(id)
+			reminder.taskName = textView.text;
+			reminder.topic = self.topic;
+			
+			textView.text = @"";
+			
+			NSError *error;
+			if (![self.managedContext save:&error]) {
+				NSLog(@"Could not save a Reminder object:%@. \n An error occured: %@", reminder, error);
+			}
+		}else {
+			
+			/*
+			 Case for Update cell
+			 We acces the cell in which the textView is in by accesing it's superview.superview
+			 Then we iterate through tableView's rows until we find the cell by computing indexPath with different row
+			 Once we find its match, we break the loop and update its content
+			 
+			 OR: Can we pass somehow the event from the textView, to it's superview, way up to the cell. And then
+			 we can get use of indexPathForSelectedRow? (like in the WWDC video with the gesture events)
+			 */
+
+			CDAddReminderCell *cellFromTextView = (CDAddReminderCell *)textView.superview.superview;
+			
+			for (int row = 0; row < ((int)self.fetchedResultsController.sections[0].numberOfObjects); row += 1) {
+				NSIndexPath *indexPath = [NSIndexPath indexPathForRow:row inSection:0];
+				if ([self.tableView cellForRowAtIndexPath:indexPath] == cellFromTextView) {
+					CDReminder *reminderToUpdate = [self.fetchedResultsController objectAtIndexPath:indexPath];
+					reminderToUpdate.taskName = textView.text;
+					
+					NSError *error;
+					if (![self.managedContext save:&error]) {
+						NSLog(@"Could not update a Reminder obj:%@ \n An error occured: %@", reminderToUpdate, error);
+					}
+					
+					break;
+				}
+			}
+		}
+		return NO;
+	}
+	
+	return YES;
+	
+}
+
+#pragma mark - UIScrollViewDelegate
 
+- (void)scrollViewWillBeginDragging:(UIScrollView *)scrollView {
+	
+	if(self.tableView == scrollView) {
+		NSIndexPath *indexPath = [NSIndexPath indexPathForRow:0 inSection:0];
+		[[self.tableView cellForRowAtIndexPath:indexPath] resignFirstResponder];
+	}
+	
+}
+
+#pragma mark - Table view data source
+
+- (CGFloat)tableView:(UITableView *)tableView heightForRowAtIndexPath:(NSIndexPath *)indexPath {
+	
+	return [self textViewHeightForRowAtIndexPath:indexPath];
+
+}
+
+- (NSInteger)numberOfSectionsInTableView:(UITableView *)tableView {
+	
+	return self.fetchedResultsController.sections.count;
+	
+}
+
+- (NSInteger)tableView:(UITableView *)tableView numberOfRowsInSection:(NSInteger)section {
+	
+	/* We want to return the number of items in the database plus a temporary cell in which the user can add
+	   another reminder when the cell its tapped */
+	return self.fetchedResultsController.sections[section].numberOfObjects + 1;
+	
 }
 
 - (UITableViewCell *)tableView:(UITableView *)tableView cellForRowAtIndexPath:(NSIndexPath *)indexPath {
 	
-	UITableViewCell *cell = [self.tableView dequeueReusableCellWithIdentifier:@"SimpleReminderCell"];
+	CDReminderCell *cell = [tableView dequeueReusableCellWithIdentifier:@"AddCell" forIndexPath:indexPath];
```

Comment:
Add logic here to show the appropriate cell for the index path. For the last index create a cell with identifier "AddCell" and for the rest create of type "ReminderCellBasic".